crates/sui-types/src/crypto.rs

```diff
@@ -135,13 +146,13 @@ impl SuiAuthoritySignature for AuthoritySignature {
         secret.sign(&message)
     }
 
-    fn verify<T>(&self, value: &T, author: PublicKeyBytes) -> Result<(), SuiError>
+    fn verify<T>(&self, value: &T, author: AuthorityPublicKeyBytes) -> Result<(), SuiError>
     where
         T: Signable<Vec<u8>>,
     {
         // is this a cryptographically valid public Key?
-        let public_key =
-            PublicKey::from_bytes(author.as_ref()).map_err(|_| SuiError::InvalidAddress)?;
+        let public_key = AuthorityPublicKey::from_bytes(author.as_ref())
+            .map_err(|_| SuiError::InvalidAddress)?;
```

Comment:
the error name seems misleading? maybe `SuiError::KeyConversionError`

Reply:
Thanks, this will be changed.